### PR TITLE
Only part of the crud that's in there ...

### PR DESCRIPTION
**Pull request by rocky**
for OS's that I doubt we'll ever support. And if we do,
it would be done in a cleaner way.

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -145,7 +145,7 @@ endif
 
 # Extra stuff to include in the distribution.
 
-mk_FILES =	Basic.mk mk/msdosdjgpp.mk mk/Amiga.mk mk/Windows32.mk
```
Comment (boretom) [inline Makefile.am:148]: @rocky got the mk/msdosdjpp.mk have anything to do with compiling for DOS?

File: README.template [hunk 2]
```diff
@@ -147,19 +147,11 @@ Ports
...
     Customs distributed build environment from the Pmake distribution.
 
-  - See README.VMS for details about GNU Make on OpenVMS.
-
-  - See README.Amiga for details about GNU Make on AmigaDOS.
-
   - See README.W32 for details about GNU Make on Windows NT, 95, or 98.
 
   - See README.DOS for compilation instructions on MS-DOS and MS-Windows
     using DJGPP tools.
```
Comment (boretom) [inline README.template:153]: If DJGG is not used to compile on Windows and MS-DOS, I'd rephrase that.

File: src/function.c [hunk 3]
```diff
@@ -554,24 +537,11 @@ func_notdir_suffix (char *o, char **argv, const char *funcname)
...
           o = variable_buffer_output (o, p, len - (p - p2));
         }
-#ifdef HAVE_DOS_PATHS
-      /* Handle the case of "d:foo/bar".  */
-      else if (is_notdir && p2[0] && p2[1] == ':')
-        {
-          p = p2 + 2;
-          o = variable_buffer_output (o, p, len - (p - p2));
-        }
-#endif
```
Comment (boretom) [inline src/function.c:564]: That is not needed for Windows / MS-DOS support?

File: src/function.c [hunk 4]
```diff
@@ -621,38 +578,13 @@ func_basename_dir (char *o, char **argv, const char *funcname)
         o = variable_buffer_output (o, p2, ++p - p2);
       else if (p >= p2 && (*p == '.'))
         o = variable_buffer_output (o, p2, p - p2);
-#ifdef HAVE_DOS_PATHS
-      /* Handle the "d:foobar" case */
-      else if (p2[0] && p2[1] == ':' && is_dir)
-        o = variable_buffer_output (o, p2, 2);
-#endif
```
Comment (boretom) [inline src/function.c:628]: Same as above..

**Review by boretom — COMMENTED** [review]
General: all the DOS directives are not used in Windows?

File: Makefile.am (hunk 1 above)
Comment (rocky) [reply Makefile.am:148]: Back in ancient days of yore possibly before you were born (circa 1989) there was this marvelous extension to DOS to make it work like a real 32-bit system and allow it to run GCC and thus compile GNU programs. See https://en.wikipedia.org/wiki/DJGPP

People who still have those systems and love it, see https://hackaday.com/2018/05/14/msdos-development-with-gcc/, should use GNU Make. 

If for some reason there is a need for that there, the code could be added back in, but it would _have_ to be done in a way that doesn't require numerous files strewn about, and `#ifdefs` all over the place. A better way to do this is to create a single directory with that code in it, and duplicate the functions that need to be modified. `#ifdef` is fine for small changes, but what we see here in `ar.c`, `arscan.c`, and `function.c` is just a bit over the top. 

In one place we can't even get the indentation of some code right because in one variant it is part of an `else` while in the bulk of the cases it isn't.  And so the indention is wrong for the bulk case so as to be correct for the obscure case.

**Comment by rocky** [comment]
Acutally Native Windows support is also kind of on the way out too. I don't test it. Do you? 

As with DOS, Amiga, and VMS, if that becomes broken _and_ there is a desire to fix it, then it would have to be done in a cleaner way.

**Comment by boretom** [comment]
I don't test on Windows, it's too much of niche product :D ... and you're sure right that it could added back in a better way. The same of course is true for VMS, Amiga and DOS.


>Back in ancient days of yore possibly before you were born

Me not having a clue is only beaten by me not having a clue and being old (not too long until 50). Came late to computers, it took it's time from being a plumber to get a special kind of high school diploma to start at university for applied science (2x, since I didn't like building services, my first choice). And then I'm mostly a sys-/app-admin with (uni) knowledge of programming.
But I do have more free time now and I just love open source and see what (little) I can contribute.

**Comment by rocky** [comment]
Hey, I think plumbing is a great activity. Just paid someone $400 to fix a sink that had a corroded drain. One of the best jobs I had was as a (mostl tin) roofer. I think that kind of stuff is fantastic!

**Review by boretom — APPROVED** [review]
Getting rid of "altlasten" is perfect.
